common/src/com/thoughtworks/go/remote/work/ConsoleOutputTransmitter.java

```diff
@@ -86,10 +94,9 @@ public void flushToServer() {
             consoleAppender.append(result.toString());
         } catch (IOException e) {
             LOGGER.warn("Could not send console output to server", e);
-            //recreate buffer
             synchronized (buffer) {
                 sent.addAll(buffer);
-                buffer = new CircularFifoBuffer(10 * 1024);
+                buffer.clear();
```

Comment:
Hi @jyotisingh, I cannot think of a reason this would impact performance or how it would be logically more correct to instantiate a new buffer object rather than clearing the existing one. It may have been unintentional.

Worse though, the code synchronizes on `buffer`, and changing the reference to a new object defeats the purpose of the mutex -- that's why I changed it back to `clear()`.